server/expense.go

```diff
@@ -58,25 +58,47 @@ type ExpenseInput struct {
 	Category ExpenseCategory `json:"category" binding:"required"`
 	Costs    string          `json:"costs" binding:"required"`
 	Date     string          `json:"date" binding:"required"`
+	TagIDs   []string        `json:"tag_id"`
+}
+
+type ExpenseOutput struct {
+	Expense Expense
+	Tags    []ExpenseTagOutput
 }
 
 func insertExpense(c *gin.Context) {
 	var newExpense ExpenseInput
-	err := c.BindJSON(&newExpense)
-	if err != nil {
+	if err := c.BindJSON(&newExpense); err != nil {
+		saveErrorInfo(c, err, 400)
 		return
 	}
 
-	/* get userID from middleware */
 	userID := c.MustGet("userID")
 
 	db := GetDB()
-	_, err = db.Exec("INSERT INTO expense VALUES (uuid_generate_v4(), $1, $2, $3, $4, $5, now(), now())", newExpense.Name, newExpense.Category, newExpense.Costs, userID, newExpense.Date)
+	rows, err := db.NamedQuery("INSERT INTO expense VALUES (uuid_generate_v4(), :name, :category, :costs, :user_id, :date, now(), now()) RETURNING id",
+		map[string]interface{}{
+			"name":     newExpense.Name,
+			"category": newExpense.Category,
+			"costs":    newExpense.Costs,
+			"user_id":  userID.(string),
+			"date":     newExpense.Date})
 
 	/* if there is a database error */
 	if err != nil {
-		log.Println(err)
-		c.AbortWithStatus(500)
+		saveErrorInfo(c, err, 500)
+		return
+	}
+	/* read back generated expense ID */
+	var expenseID string
+	if rows.Next() {
```

Comment:
If you are really sure this is never null, you can remove the if

Reply:
Does not work. rows.scan() delivers nothing